quaternions/quaternions.py

```diff
@@ -293,21 +293,75 @@ def from_qmethod(source, target, probabilities=None):
         return Quaternion._first_eigenvector(K)
 
     @staticmethod
-    def average(*quaternions, weights=None):
+    def average(*quaternions, weights=()):
         '''
         Return the quaternion such that its matrix minimizes the square distance
         to the matrices of the quaternions in the argument list.
 
         See Averaging Quaternions, by Markley, Cheng, Crassidis, Oschman.
         '''
         B = np.array([q.coordinates for q in quaternions])
-        if not weights:
+        if not any(True for _ in weights):
             weights = np.ones(len(quaternions))
         M = B.T.dot(np.diag(weights)).dot(B)
 
         return Quaternion._first_eigenvector(M)
 
     @staticmethod
+    def average_and_std_naive(*quaternions, weights=()):
+        """
+        Naive implementation of std. dev. calculation, assuming small angles between quaternions
+        and average. Returns average quaternion and stddev of input quaternion list in deg
+        """
+        if not any(True for _ in weights):
+            weights = np.ones(len(quaternions))
+        q_average = Quaternion.average(*quaternions, weights=weights)
+        diffs = np.array([q_average.distance(quat) for quat in quaternions])
+        # TODO: check that this way of taking into account weights in stddev calculation is ok
+        stddev = np.degrees(np.sqrt(sum((diffs ** 2) * weights) / np.sum(weights)))
+        return q_average, stddev
+
+    @staticmethod
+    def average_and_std_lerner(*quaternions, weights=()):
+        """
+        Calculates the std. dev. as 1.87 of the root of the maximum eigenvalue of the covariance matrix,
```

Comment:
I am not an expert on quaternions, but Google didn't show any relevant results on `"quaternions" "sigma lerner"` and variations. Perhaps it's coming from some article by Gerald M. Lerner? Is there a chance we can document this a bit more?

Reply:
certainly, i have to track down the paper from which this way of condensing a correlation matrix to a single value was taken...